### DISCUSSION TITLE
Use modal sheet instead of dismissable ImageGallery

### DISCUSSION THREAD

**Discussion by Cierra-Runis** [flyerhq/flutter_chat_ui#526] [Q&A] [answered]
Since dismissable ImageGallery doesn't pushed to navigation, we can't use Android system back gesture to dismiss ImageGallery.
By using modal sheet, it can pop with back gesture and remove code that managing states of ImageGallery.

**Comment by demchenkoalex** [comment]
is it okay to include navigation/modals inside of the library? will it work nicely with any flutter application?

**Reply by Cierra-Runis** [reply]
Most of flutter projects are using `MaterialApp`, which need import material library, and [showModalBottomSheet](https://api.flutter.dev/flutter/material/showModalBottomSheet.html) method is include in it.

If showModalBottomSheet method is not good for you, just use `Navigator.push` because Navigator must exist, or give an optional function for developer to handle it.

**Reply by demchenkoalex** [reply]
Well bottom sheet is not really suitable for the image preview, is it? Never ever saw image preview in a bottom sheet :) 

I think I will just add some callbacks for library users to manage this, cause I don't feel super comfortable with navigating from the UI library. The image preview library is also kinda meh so I will allow for people to use their implementations.

**Comment by demchenkoalex** [comment, marked as answer]
image preview is dropped in v2 https://github.com/flyerhq/flutter_chat_ui/issues/646 due to the quality of the package used. Maybe eventually I will write yet another package for image preview specifically, but probably not soon.
